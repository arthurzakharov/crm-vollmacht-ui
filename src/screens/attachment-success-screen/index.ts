export * from "./attachment-success-screen";
