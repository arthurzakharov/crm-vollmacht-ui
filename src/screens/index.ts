export * from "./attachment-success-screen";
export * from "./error-screen";
export * from "./home-screen";
export * from "./loading-screen";
