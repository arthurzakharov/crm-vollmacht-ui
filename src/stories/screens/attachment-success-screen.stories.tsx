import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { AttachmentSuccessScreen } from "../../screens";
import logoSrc from "../assets/logo.png";
import submittedSrc from "../assets/submitted.png";

type Story = StoryObj<typeof AttachmentSuccessScreen>;

const meta: Meta<typeof AttachmentSuccessScreen> = {
  title: "SCREENS/HomeScreen",
  component: AttachmentSuccessScreen,
  args: {
    header: {
      logo: logoSrc,
      tel: "030 / 20 898 12 11",
      descriptions: [
        {
          size: "m",
          value: "Mo-Fr von 8:00 - 20:00",
        },
      ],
    },
    main: {
      screen: "attachment",
      title: "Verg√ºtungsvereinbarung und Vollmacht",
    },
    footer: {
      name: "legal-one",
      links: [
        {
          text: "Allgemeine Informationen",
          onClick: fn(),
        },
        {
          text: "Datenschutz",
          onClick: fn(),
        },
        {
          text: "Impressum",
          onClick: fn(),
        },
      ],
    },
    success: {
      icon: submittedSrc,
      title: "Vielen Dank!",
      text: [
        "Wir haben Ihre Informationen erhalten und werden diese umgehend bearbeiten.",
        "Sobald neue Erkentnisse vorliegen, setzen wir uns mit Ihnen in Verbindung.",
      ],
    },
  },
};

export const Default: Story = {
  name: "Attachment Success Screen",
};

export default meta;
