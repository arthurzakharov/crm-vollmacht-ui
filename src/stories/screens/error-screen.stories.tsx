import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { ErrorScreen } from "../../screens";
import logoSrc from "../assets/logo.png";

type Story = StoryObj<typeof ErrorScreen>;

const meta: Meta<typeof ErrorScreen> = {
  title: "SCREENS/ErrorScreen",
  component: ErrorScreen,
};

export const Default: Story = {
  name: "Error Screen",
  args: {
    header: {
      logo: logoSrc,
      tel: "030 / 20 898 12 11",
      descriptions: [
        {
          size: "m",
          value: "Mo-Fr von 8:00 - 20:00",
        },
      ],
    },
    footer: {
      name: "legal-one",
      links: [
        {
          text: "Allgemeine Informationen",
          onClick: fn(),
        },
        {
          text: "Datenschutz",
          onClick: fn(),
        },
        {
          text: "Impressum",
          onClick: fn(),
        },
      ],
    },
    error: {
      title: "Entschuldigung, da ist wohl etwas schief gelaufen!",
      subtitle: "Fehler 404",
      tableTitle: "Bitte kontaktieren Sie uns doch per:",
      tableRows: [
        {
          key: "E-Mail:",
          value: "info@rightmart.de",
        },
        {
          key: "Telefon:",
          value: "+49 (0)421 / 33 100 310",
        },
        {
          key: "Fax:",
          value: "+49 (0)421 / 33 100 380",
        },
        {
          key: "Post:",
          value: "rightmart Rechtsanwaltsgesellschaft mbH<br />Clara-Jaschke-Stra√üe 1<br />28199 Bremen",
        },
      ],
    },
  },
};

export default meta;
