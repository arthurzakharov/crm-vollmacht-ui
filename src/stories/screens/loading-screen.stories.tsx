import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { LoadingScreen } from "../../screens";
import logoSrc from "../assets/logo.png";

type Story = StoryObj<typeof LoadingScreen>;

const meta: Meta<typeof LoadingScreen> = {
  title: "SCREENS/Loading Screen",
  component: LoadingScreen,
};

export const Default: Story = {
  name: "Loading Screen",
  args: {
    header: {
      logo: logoSrc,
      tel: "030 / 20 898 12 11",
      descriptions: [
        {
          size: "m",
          value: "Mo-Fr von 8:00 - 20:00",
        },
      ],
    },
    footer: {
      name: "legal-one",
      links: [
        {
          text: "Allgemeine Informationen",
          onClick: fn(),
        },
        {
          text: "Datenschutz",
          onClick: fn(),
        },
        {
          text: "Impressum",
          onClick: fn(),
        },
      ],
    },
  },
};

export default meta;
