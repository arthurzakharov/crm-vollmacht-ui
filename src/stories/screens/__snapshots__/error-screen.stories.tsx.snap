// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCREENS/Error Screen Default smoke-test 1`] = `
<div class="layout-without-sidebar">
  <div class="layout-without-sidebar__header">
    <header class="header">
      <img alt="logo-icon"
           src="/src/stories/assets/logo.png"
           class="header__logo"
      >
      <div class="header__info">
        <button tabindex="0"
                class="header__link"
        >
          <img alt="phone-icon"
               src="/src/assets/png/phone/phone.png"
               class="header__phone"
          >
          <span class="header__number">
            030 / 20 898 12 11
          </span>
        </button>
        <div class="header__description header__description--m">
          <svg focusable="false"
               role="img"
               xmlns="http://www.w3.org/2000/svg"
               viewbox="0 0 512 512"
          >
            <path fill="currentColor"
                  d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 48c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m140.204 130.267l-22.536-22.718c-4.667-4.705-12.265-4.736-16.97-.068L215.346 303.697l-59.792-60.277c-4.667-4.705-12.265-4.736-16.97-.069l-22.719 22.536c-4.705 4.667-4.736 12.265-.068 16.971l90.781 91.516c4.667 4.705 12.265 4.736 16.97.068l172.589-171.204c4.704-4.668 4.734-12.266.067-16.971z"
            >
            </path>
          </svg>
          <span>
            Mo-Fr von 8:00 - 20:00
          </span>
        </div>
      </div>
    </header>
  </div>
  <section class="error-404">
    <h1 class="error-404__h1">
      Entschuldigung, da ist wohl etwas schief gelaufen!
    </h1>
    <h2 class="error-404__h2">
      Fehler 404
    </h2>
    <h3 class="error-404__h3">
      Bitte kontaktieren Sie uns doch per:
    </h3>
    <table class="error-404__table">
      <tbody>
        <tr>
          <th class="error-404__th">
            E-Mail:
          </th>
          <td class="error-404__td">
            info@rightmart.de
          </td>
        </tr>
        <tr>
          <th class="error-404__th">
            Telefon:
          </th>
          <td class="error-404__td">
            +49 (0)421 / 33 100 310
          </td>
        </tr>
        <tr>
          <th class="error-404__th">
            Fax:
          </th>
          <td class="error-404__td">
            +49 (0)421 / 33 100 380
          </td>
        </tr>
        <tr>
          <th class="error-404__th">
            Post:
          </th>
          <td class="error-404__td">
            rightmart Rechtsanwaltsgesellschaft mbH
            <br>
            Clara-Jaschke-Straße 1
            <br>
            28199 Bremen
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <div class="layout-without-sidebar__footer">
    <footer class="footer">
      <hr class="footer__line">
      <div class="footer__content">
        <span>
          © 2024 legal-one
        </span>
        <ul class="footer__links">
          <li class="footer__link">
            <button type="button"
                    tabindex="0"
            >
              Allgemeine Informationen
            </button>
            <div class="footer__separator">
            </div>
          </li>
          <li class="footer__link">
            <button type="button"
                    tabindex="0"
            >
              Datenschutz
            </button>
            <div class="footer__separator">
            </div>
          </li>
          <li class="footer__link">
            <button type="button"
                    tabindex="0"
            >
              Impressum
            </button>
            <div class="footer__separator">
            </div>
          </li>
        </ul>
      </div>
    </footer>
  </div>
</div>
`;
